src/main/java/seedu/address/model/poc/UniquePocList.java

```diff
@@ -0,0 +1,138 @@
+import seedu.address.model.poc.exceptions.DuplicatePocException;
+import seedu.address.model.poc.exceptions.PocNotFoundException;
+
+/**
+ * A list of companies that enforces uniqueness between its elements and does not allow nulls.
+ * A company is considered unique by comparing using {@code Company#isSameCompany(Company)}.
+ * As such, adding and updating of companies uses Company#isSameCompany(Company) for equality to ensure that the
+ * company being added or updated is unique in terms of identity in the UniqueCompanyList.
+ * However, the removal of a company uses Company#equals(Object) to ensure that the company with exactly
+ * the same fields will be removed.
+ *
+ * Supports a minimal set of list operations.
+ *
+ * @see Poc#isSamePoc(Poc)
+ */
+public class UniquePocList implements Iterable<Poc> {
+
+    private final ObservableList<Poc> internalList = FXCollections.observableArrayList();
+    private final ObservableList<Poc> internalUnmodifiableList =
+            FXCollections.unmodifiableObservableList(internalList);
+
+    /**
+     * Returns true if the list contains an equivalent company as the given argument.
+     */
+    public boolean contains(Poc toCheck) {
+        requireNonNull(toCheck);
+        return internalList.stream().anyMatch(toCheck::isSamePoc);
+    }
+
+    /**
+     * Adds a company to the list.
+     * The company must not already exist in the list.
+     */
+    public void add(Poc toAdd) {
+        requireNonNull(toAdd);
+        if (contains(toAdd)) {
+            throw new DuplicatePocException();
+        }
+        internalList.add(toAdd);
+    }
+
+    /**
+     * Replaces the company {@code target} in the list with {@code editedCompany}.
+     * {@code target} must exist in the list.
+     * The company identity of {@code editedCompany} must not be the same as another existing company in the list.
+     */
+    public void setPoc(Poc target, Poc editedPoC) {
+        requireAllNonNull(target, editedPoC);
+
+        int index = internalList.indexOf(target);
+        if (index == -1) {
+            throw new PocNotFoundException();
+        }
+
+        if (!target.isSamePoc(editedPoC) && contains(editedPoC)) {
+            throw new DuplicatePocException();
+        }
+
+        internalList.set(index, editedPoC);
+    }
+
+    /**
+     * Removes the equivalent company from the list.
+     * The company must exist in the list.
+     */
+    public void remove(Poc toRemove) {
+        requireNonNull(toRemove);
+        if (!internalList.remove(toRemove)) {
+            throw new PocNotFoundException();
+        }
+    }
+
+    public void setPocs(UniquePocList replacement) {
+        requireNonNull(replacement);
+        internalList.setAll(replacement.internalList);
+    }
+
+    /**
+     * Replaces the contents of this list with {@code companies}.
+     * {@code companies} must not contain duplicate companies.
+     */
+    public void setPocs(List<Poc> pocs) {
```

Comment:
Perhaps could consider renaming this method to replacePocs to prevent potential confusion from the previous method setPoc